test/unit/util/release/changelog_test.py

```diff
@@ -0,0 +1,123 @@
+from inspect import cleandoc
+
+import pytest
+
+from exasol.toolbox.util.release.changelog import (
+    UNRELEASED_TEXT,
+    Changelogs,
+)
+from exasol.toolbox.util.version import Version
+
+CHANGES_CONTENTS = cleandoc(
+    """
+    # Changelog
+
+    * [unreleased](unreleased.md)
+    * [0.1.0](changes_0.1.0.md)
+
+    ```{toctree}
+    ---
+    hidden:
+    ---
+    unreleased
+    changes_0.1.0
+    ```
+    """
+)
+
+ALTERED_CHANGES_CONTENTS = cleandoc(
+    """
+    # Changelog
+
+    * [unreleased](unreleased.md)
+    * [1.0.0](changes_1.0.0.md)
+    * [0.1.0](changes_0.1.0.md)
+
+    ```{toctree}
+    ---
+    hidden:
+    ---
+    unreleased
+    changes_1.0.0
+    changes_0.1.0
+    ```
+    """
+)
+
+SHARED_TEXT = cleandoc(
+    """
+    ## Added
+    * Added Awesome feature
+
+    ## Changed
+    * Some behaviour
+
+    ## Fixed
+    * Fixed nasty bug
+    """
+)
+
+
+@pytest.fixture
+def changes_md(tmp_path, changelogs):
+    """
+    As some operations in Changelogs modify files, we need a reset per function
```

Comment:
```suggestion
    As some methods in class `Changelogs` modify files, The tests need to reset this fixture per function.
```

Reply:
Maybe consider
* writing this sentence only once on top of the fixtures with function scope
* adding scope="function" explicitly